Microsoft.SystemForCrossDomainIdentityManagement/Protocol/ProtocolExtensions.cs

```diff
@@ -102,90 +103,192 @@ private static void Apply(this Core2Group group, PatchOperation2 operation)
-                                IEnumerable<Member> newMembers =
-                                     operation
-                                     .Value
-                                     .Select(
-                                         (OperationValue item) =>
-                                             new Member()
-                                             {
-                                                 Value = item.Value
-                                             })
-                                     .ToArray();
-                                group.Members =
-                                    group.Members != null ?
-                                        group.Members.Concat(newMembers).ToArray() : newMembers;
-
-                                break;
-
-                            case OperationName.Remove:
-                                if (null == group.Members)
-                                {
-                                    break;
-                                }
-
-                                if (operation?.Value?.FirstOrDefault()?.Value == null)
-                                {
-                                    group.Members = Enumerable.Empty<Member>();
-                                    break;
-                                }
-
-                                IDictionary<string, Member> members =
-                                    new Dictionary<string, Member>(group.Members.Count());
-                                foreach (Member item in group.Members)
-                                {
-                                    members.Add(item.Value, item);
-                                }
-
-                                foreach (OperationValue operationValue in operation.Value)
-                                {
-                                    if (members.TryGetValue(operationValue.Value, out Member removedMember))
-                                    {
-                                        members.Remove(operationValue.Value);
-                                    }
-                                }
-
-                                group.Members = members.Values;
-                                break;
-                        }
+                        members.Remove(operationValue.Value);
                     }
-                    break;
+                }
+            }
+
+            //we have paths we need to evaluate and remove
+            if (operation.Path?.SubAttributes?.Count == 1)
+            {
+                members.RemoveMembersOperationByPath(operation);
+            }
+
+            group.Members = members.Values;
+        }
+
+        private static void RemoveMembersOperationByPath(this Dictionary<string, Member> memberDictionary, PatchOperation2 operation)
+        {
+            var subAttr = operation.Path.SubAttributes.Single();
+
+            if (!subAttr.AttributePath.Equals("value", StringComparison.Ordinal))
```

Comment:
No magic strings. There is a variable that should have it's value as "value" (AttributeNames.Value).